cmd/bucket-replication.go

```diff
@@ -1359,12 +1359,16 @@ func (ri ReplicateObjectInfo) replicateAll(ctx context.Context, objectAPI Object
 			if err := replicateObjectWithMultipart(ctx, c, tgt.Bucket, object,
 				r, objInfo, putOpts); err != nil {
 				rinfo.ReplicationStatus = replication.Failed
```

Comment:
why is replicationStatus failed for a good case @poornas ? isn't PrecondtionFailed a good scenario i.e the object is already replicated?

Reply:
updated PR - we could just mark it completed if PreConditionFailed